# Exploracion y decisiones

In [40]:
import sys, pandas as pd
from pathlib import Path

In [13]:
DATA_PATH = Path('../data/raw/creditcard.csv')

df = pd.read_csv(DATA_PATH)
print(f'Shape: {df.shape}')
display(df.head())

Shape: (284807, 31)


Time        V1        V2        V3        V4        V5        V6        V7  \
0   0.0 -1.359807 -0.072781  2.536347  1.378155 -0.338321  0.462388  0.239599   
1   0.0  1.191857  0.266151  0.166480  0.448154  0.060018 -0.082361 -0.078803   
2   1.0 -1.358354 -1.340163  1.773209  0.379780 -0.503198  1.800499  0.791461   
3   1.0 -0.966272 -0.185226  1.792993 -0.863291 -0.010309  1.247203  0.237609   
4   2.0 -1.158233  0.877737  1.548718  0.403034 -0.407193  0.095921  0.592941   

         V8        V9  ...       V21       V22       V23       V24       V25  \
0  0.098698  0.363787  ... -0.018307  0.277838 -0.110474  0.066928  0.128539   
1  0.085102 -0.255425  ... -0.225775 -0.638672  0.101288 -0.339846  0.167170   
2  0.247676 -1.514654  ...  0.247998  0.771679  0.909412 -0.689281 -0.327642   
3  0.377436 -1.387024  ... -0.108300  0.005274 -0.190321 -1.175575  0.647376   
4 -0.270533  0.817739  ... -0.009431  0.798278 -0.137458  0.141267 -0.206010   

        V26       V27       V28  Amount  Class  
0 -0.189115  0.133558 -0.021053  149.62      0  
1  0.125895 -0.008983  0.014724    2.69      0  
2 -0.139097 -0.055353 -0.059752  378.66      0  
3 -0.221929  0.062723  0.061458  123.50      0  
4  0.502292  0.219422  0.215153   69.99      0  

[5 rows x 31 columns]

In [26]:
display(df.dtypes)

Time      float64
V1        float64
V2        float64
V3        float64
V4        float64
V5        float64
V6        float64
V7        float64
V8        float64
V9        float64
V10       float64
V11       float64
V12       float64
V13       float64
V14       float64
V15       float64
V16       float64
V17       float64
V18       float64
V19       float64
V20       float64
V21       float64
V22       float64
V23       float64
V24       float64
V25       float64
V26       float64
V27       float64
V28       float64
Amount    float64
Class       int64
dtype: object

In [42]:
nulls = df.isna().sum().sort_values(ascending=False)
print(f'Nulos: {nulls[nulls > 0]}')

duplicates = df.duplicated().sum()
print(f'Duplicados: {duplicates}')
display(df[df.duplicated(keep=False)].head(10))

Nulos: Series([], dtype: int64)
Duplicados: 1081


Time        V1        V2        V3        V4        V5        V6  \
32    26.0 -0.529912  0.873892  1.347247  0.145457  0.414209  0.100223   
33    26.0 -0.529912  0.873892  1.347247  0.145457  0.414209  0.100223   
34    26.0 -0.535388  0.865268  1.351076  0.147575  0.433680  0.086983   
35    26.0 -0.535388  0.865268  1.351076  0.147575  0.433680  0.086983   
112   74.0  1.038370  0.127486  0.184456  1.109950  0.441699  0.945283   
113   74.0  1.038370  0.127486  0.184456  1.109950  0.441699  0.945283   
114   74.0  1.038370  0.127486  0.184456  1.109950  0.441699  0.945283   
115   74.0  1.038370  0.127486  0.184456  1.109950  0.441699  0.945283   
220  145.0 -2.420413  1.947885  0.553646  0.983069 -0.281518  2.408958   
221  145.0 -2.420413  1.947885  0.553646  0.983069 -0.281518  2.408958   

           V7        V8        V9  ...       V21       V22       V23  \
32   0.711206  0.176066 -0.286717  ...  0.046949  0.208105 -0.185548   
33   0.711206  0.176066 -0.286717  ...  0.046949  0.208105 -0.185548   
34   0.693039  0.179742 -0.285642  ...  0.049526  0.206537 -0.187108   
35   0.693039  0.179742 -0.285642  ...  0.049526  0.206537 -0.187108   
112 -0.036715  0.350995  0.118950  ...  0.102520  0.605089  0.023092   
113 -0.036715  0.350995  0.118950  ...  0.102520  0.605089  0.023092   
114 -0.036715  0.350995  0.118950  ...  0.102520  0.605089  0.023092   
115 -0.036715  0.350995  0.118950  ...  0.102520  0.605089  0.023092   
220 -1.401613 -0.188299  0.675878  ...  1.213826 -1.238620  0.006927   
221 -1.401613 -0.188299  0.675878  ...  1.213826 -1.238620  0.006927   

          V24       V25       V26       V27       V28  Amount  Class  
32   0.001031  0.098816 -0.552904 -0.073288  0.023307    6.14      0  
33   0.001031  0.098816 -0.552904 -0.073288  0.023307    6.14      0  
34   0.000753  0.098117 -0.553471 -0.078306  0.025427    1.77      0  
35   0.000753  0.098117 -0.553471 -0.078306  0.025427    1.77      0  
112 -0.626463  0.479120 -0.166937  0.081247  0.001192    1.18      0  
113 -0.626463  0.479120 -0.166937  0.081247  0.001192    1.18      0  
114 -0.626463  0.479120 -0.166937  0.081247  0.001192    1.18      0  
115 -0.626463  0.479120 -0.166937  0.081247  0.001192    1.18      0  
220 -1.724222  0.239603 -0.313703 -0.188281  0.119831    6.00      0  
221 -1.724222  0.239603 -0.313703 -0.188281  0.119831    6.00      0  

[10 rows x 31 columns]

In [35]:
display(df["Class"].value_counts(normalize=True))

Class
0    0.998273
1    0.001727
Name: proportion, dtype: float64

In [37]:
display(df[["Time", "Amount"]].describe())

Time         Amount
count  284807.000000  284807.000000
mean    94813.859575      88.349619
std     47488.145955     250.120109
min         0.000000       0.000000
25%     54201.500000       5.600000
50%     84692.000000      22.000000
75%    139320.500000      77.165000
max    172792.000000   25691.160000

In [41]:
print(f'Version Python: {sys.version}')
print(f'Version Pandas: {pd.__version__}')

Version Python: 3.10.12 (main, Nov  4 2025, 08:48:33) [GCC 11.4.0]
Version Pandas: 2.3.3


## Conslusion

### Descripcion del dataset

Dataset de transacciones financieras con 284.807 registros y 31 columnas.
No contiene timestamps explicitos, solo una variable Time relativa.

### Problemas
* No hay una fecha real
* Dataset muy desbalanceado (Class) 99.82% Vs. 0.17%
* Hay 1081 registros duplicados
    * Los duplicados aparecen en filas consecutivas, por lo que pueden ser transacciones que esten durando mas de 1 segundo, o que se hagan justo terminando un segundo y salga en ambos.

### Decisiones
* No disponemos de un identificador de transacción ni de semántica temporal más fina, por lo que se asume que filas completamente idénticas representan duplicados técnicos.
* Se creara columna time_ts a partir de la columna Time
* Se metera ingestion_date para particionado
* Formato de salida: Parquet